pkg/plugins/policies/core/rules/resourcerules.go

```diff
@@ -0,0 +1,172 @@
+package rules
+
+import (
+	"slices"
+
+	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	meshextenralservice_api "github.com/kumahq/kuma/pkg/core/resources/apis/meshexternalservice/api/v1alpha1"
+	meshservice_api "github.com/kumahq/kuma/pkg/core/resources/apis/meshservice/api/v1alpha1"
+	core_model "github.com/kumahq/kuma/pkg/core/resources/model"
+	meshhttproute_api "github.com/kumahq/kuma/pkg/plugins/policies/meshhttproute/api/v1alpha1"
+)
+
+type ResourceRule struct {
+	Resource core_model.ResourceMeta
+	Conf     interface{}
+	Origin   []core_model.ResourceMeta
+}
+
+type UniqueResourceKey string
+
+type ResourceRules map[UniqueResourceKey]ResourceRule
+
+func (rr ResourceRules) Compute(r core_model.Resource, l ResourceLister) *ResourceRule {
+	key := uniqueKey(r)
+	if rule, ok := rr[key]; ok {
+		return &rule
+	}
+
+	findMesh := func(meshName string) core_model.Resource {
+		meshes := l.ListOrEmpty(core_mesh.MeshType).GetItems()
```

Comment:
Can't we just get mesh by key from store? We always use `ResourceLister` in this and we end up iterating over all resources, it does not look optimal from performance point of view

Reply:
This will be especially problematic when we start adding support for routes and we end up iterating over every MeshService each time we run Compute and other code here

Reply:
Under the hood it's using `MeshContext.Resources` that stores resources as `map[core_model.ResourceType]core_model.ResourceList`. Changing this to map by ResourceIdentifier can potentially affect multiple places and I'd prefer doing this in a separate PR. 

Though in this PR I created a method `Get()` that's used everywhere in `resourcerule.go` so we don't have linear searches all-over the code.

Reply:
I didn't check what we use under the Lister interface, but with MeshContext this makes sense.